Assets/Plugins/Source/Editor/Internal/BuildPackage.cs

```diff
@@ -0,0 +1,226 @@
+    /// directory in the root of the project folder named "Build"
+    /// </summary>
+    private const string DEFAULT_OUTPUT_DIRECTORY = "Build";
+
+    /// <summary>
+    /// Note that this is a Non-Unity defined command line argument.
+    /// </summary>
+    private const string ARG_FLAG_OUTPUT = "-EOSPluginOutput";
+
+    private const string ARG_FLAG_PACKAGE_TYPE = "-PackageType";
+
+    /// <summary>
+    /// Returns a path to output the plugin to.
+    /// </summary>
+    /// <returns>Path to output the plugin to</returns>
+    private static string GetOutputPath()
+    {
+        string output_path = GetCLIArgument(ARG_FLAG_OUTPUT);
+        
+        if (string.Empty == output_path)
+        {
+            // In this case we want the absolute path, because we don't
+            // want any code down-stream to try and use it as a relative path.
+            // Note that it's resolved using the project's root directory.
+            output_path = Path.GetFullPath(
+                Path.Combine(
+                    Application.dataPath,
+                    "..",
+                    DEFAULT_OUTPUT_DIRECTORY));
+        }
+
+        return output_path;
+    }
+
+    private static string GetCLIArgument(string flag)
+    {
+        string value = string.Empty;
+        string[] args = System.Environment.GetCommandLineArgs();
+        for (int i = 0; i < args.Length; ++i)
+        {
+            if (ARG_FLAG_OUTPUT == args[i] && i + 1 < args.Length)
+            {
+                value = args[i + 1];
+            }
+        }
+
+        return value;
+    }
+
+    /// <summary>
+    /// This is the JSON file that defines which files to put into the UPM directory.
+    /// @TODO: Handle the scenario where whomever is developing the UPM wants to 
+    /// introduce a new file. Is there a way that we could intelligently include that
+    /// and add it to this JSON without things getting really messy?
+    /// </summary>
+    /// <param name="type">The type of the package</param>
+    /// <returns>Filepath to the appropriate package.json file to use.</returns>
+    private static string GetDefaultJSONPackageFile(PackageType type)
+    {
+        string path = string.Empty;
+
+        switch(type)
+        {
+            case PackageType.DotUnity:
+                path = "PackageDescriptionConfigs/eos_dotunitypackage_package_desc.json";
+                break;
+            case PackageType.UPMTarBall:
+                path = "PackageDescriptionConfigs/eos_package_description.json";
+                break;
+            case PackageType.UPMDirectory:
+            default:
+                path = "PackageDescriptionConfigs/eos_export_assets_package_desc.json";
+                break;
+        }
+
+        return path;
+    }
+
+    private static PackageType GetPackageType()
+    {
+        return PackageType.UPMDirectory;
+    }
+    private static void Clean()
```

Comment:
really minor nit: needs a space

Reply:
I can't believe my OCD brain didn't hit on this. Good catch.